### PR TITLE
[keymgr/dv] Add testplan

### PR DESCRIPTION
**Pull request by weicaiyang**
Add preliminary testplan, will update when design reaches D1

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            entered,  when KEY_VERSION > MAX_*_VER
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_kmac_data
+      desc: '''
+            Use all 0s or 1s as KMAC input data
+            Check err_code INVALID_KMAC_DATA triggered and KMAC key output is disabled
```
Comment (sriyerg) [inline hw/ip/keymgr/doc/keymgr_testplan.hjson:113]: Any alert / intr triggered by this?

**Review by sriyerg — COMMENTED** [review]
Just skimmed through - I haven't gone through the keymgr spec to understand in more detail, so I will defer to @tjaychen in case anything seems missing. 

It will also be good if you could follow the same 'Goal', 'Stimulus', 'Checks' format for all entries.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            Directly go to `StDisabled` from any state
+            Check `StDisabled` is entered correctly'''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: lc_disable
+      desc: '''
+            Test life cycle disables keymgr in any state
+            Check KMAC key output is disabled
```
[the remark this answers is not in the document]
Comment (weicaiyang) [reply hw/ip/keymgr/doc/keymgr_testplan.hjson]: yes, design and spec has been updated for lc_disable recently. updated the testplan.

**Comment by weicaiyang** [comment]
> Just skimmed through - I haven't gone through the keymgr spec to understand in more detail, so I will defer to @tjaychen in case anything seems missing.
> 
> It will also be good if you could follow the same 'Goal', 'Stimulus', 'Checks' format for all entries.

Updated as you suggested. PTAL and will have a review meeting to go over this too.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            - Check alert `operation_err` is triggered and err_code is `INVALID_KMAC_INPUT`
+            - Check KMAC output key is corrupted and working state remains the same
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_kmac_data
+      desc: '''
+            Stimulus:
```
Comment (sriyerg) [inline hw/ip/keymgr/doc/keymgr_testplan.hjson:160]:
```suggestion
            Verify `keymgr` behavior with invalid data patterns. 
            
            Stimulus:
```

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            Checks:
+            Same as above entry - "invalid_cmd"
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_kmac_input
+      desc: '''
+            Stimulus:
```
Comment (sriyerg) [inline hw/ip/keymgr/doc/keymgr_testplan.hjson:145]:
```suggestion
            Verify `keymgr` behavior with invalid key version. 
            
            Stimulus:
```

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            - Check KMAC output key and output data to SW are corrupted, also keymgr enters
+              StDisabled
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: kmac_error
+      desc: '''
+            KMAC can respond error to trigger an `INVALID_CMD` error when keymgr sends data to it
```
Comment (sriyerg) [inline hw/ip/keymgr/doc/keymgr_testplan.hjson:131]:
```suggestion
            Verify `keymgr` behavior on error response received from `KMAC` when attempting to send data to it.  
```

The goal indicates what is tested very briefly at a high level. What exactly is expected to happen is covered in 'checks' part.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            - If keymgr in disabled state, check the behavior is consistent with normal behavior.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_cmd
+      desc: '''
+            Key manager has some security checks for fault injection, and it triggers an
+            `INVALID_CMD` error. This isn't possible to be raised by SW.
```
Comment (sriyerg) [inline hw/ip/keymgr/doc/keymgr_testplan.hjson:112]:
```suggestion
            Verify `keymgr` security countermeasures on fault injection.
```

The stimulus part explains why that this is done using forces. Checks part explains what countermeasures are expected.

**Review by sriyerg — APPROVED** [review]
LGTM. 

I have suggested some more edits to make it more clear what should be captured in 'goals, 'stimulus' and 'checks'.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson (hunk 6 above)
Comment (weicaiyang) [reply hw/ip/keymgr/doc/keymgr_testplan.hjson]: Thanks for the suggestion. Updated.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson (hunk 5 above)
Comment (weicaiyang) [reply hw/ip/keymgr/doc/keymgr_testplan.hjson]: Updated.

File: hw/ip/keymgr/doc/keymgr_testplan.hjson (hunk 4 above)
Comment (weicaiyang) [reply hw/ip/keymgr/doc/keymgr_testplan.hjson]: Done

File: hw/ip/keymgr/doc/keymgr_testplan.hjson (hunk 3 above)
Comment (weicaiyang) [reply hw/ip/keymgr/doc/keymgr_testplan.hjson]: Done

**Pull request merged**